### PR TITLE
[rocm6.4_internal_testing] merge miopen bf16/fp16 mixed batchnorm

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
bf16/fp16 NHWC/NCHW mixed batchnorm merged
refactoring of miopen batchnorm tests:

- `test_batchnorm_nhwc_miopen` - removed
- `test_batchnorm_layout_*_mixed_*` - added to cover all batchnorm train configs
> - `test_batchnorm_layout_NCHW_mixed_True_bfloat16` - skipped due to MIOpen tolerance issue SWDEV-507600
- `test_batchnorm_nhwc_eval_*` - added to cover all NHWC batchnorm inference configs
> - test_batchnorm_nhwc_eval_mixed_False_float32 - skipped due to SolverNotFound SWDEV-509640
> - test_batchnorm_nhwc_eval_mixed_True_bfloat16  - skipped due to SolverNotFound SWDEV-509640
> - test_batchnorm_nhwc_eval_mixed_True_float16  - skipped due to SolverNotFound SWDEV-509640

```
python test_nn.py  -v -k test_batchnorm_layout

Running tests...
----------------------------------------------------------------------
  test_batchnorm_layout_NCHW_mixed_False_bfloat16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.003s)
  test_batchnorm_layout_NCHW_mixed_False_float16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.002s)
  test_batchnorm_layout_NCHW_mixed_False_float32 (__main__.TestNN) ... ok (0.372s)
  test_batchnorm_layout_NCHW_mixed_True_bfloat16 (__main__.TestNN) ... skip: MIOpen tolerance issue for NCHW BF16 mixed batchnorm SWDEV-507600 (0.002s)
  test_batchnorm_layout_NCHW_mixed_True_float16 (__main__.TestNN) ... ok (0.113s)
  test_batchnorm_layout_NCHW_mixed_True_float32 (__main__.TestNN) ... skip: mixed precision is useless for float32 (0.002s)
  test_batchnorm_layout_NHWC_mixed_False_bfloat16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.002s)
  test_batchnorm_layout_NHWC_mixed_False_float16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.002s)
  test_batchnorm_layout_NHWC_mixed_False_float32 (__main__.TestNN) ... ok (0.105s)
  test_batchnorm_layout_NHWC_mixed_True_bfloat16 (__main__.TestNN) ... ok (0.102s)
  test_batchnorm_layout_NHWC_mixed_True_float16 (__main__.TestNN) ... ok (0.106s)
  test_batchnorm_layout_NHWC_mixed_True_float32 (__main__.TestNN) ... skip: mixed precision is useless for float32 (0.003s)
----------------------------------------------------------------------
Ran 12 tests in 0.817s

OK (skipped=7)
```

```
python test_nn.py -v -k test_batchnorm_nhwc_eval

Running tests...
----------------------------------------------------------------------
  test_batchnorm_nhwc_eval_mixed_False_bfloat16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.002s)
  test_batchnorm_nhwc_eval_mixed_False_float16 (__main__.TestNN) ... skip: pure mode not supported for bf16/fp16 on ROCm (0.001s)
  test_batchnorm_nhwc_eval_mixed_False_float32 (__main__.TestNN) ... skip: MIOpen SolverNotFound for FP32/Fp16/BF16 NHWC batchnorm SWDEV-509640 (0.001s)
  test_batchnorm_nhwc_eval_mixed_True_bfloat16 (__main__.TestNN) ... skip: MIOpen SolverNotFound for FP32/Fp16/BF16 NHWC batchnorm SWDEV-509640 (0.001s)
  test_batchnorm_nhwc_eval_mixed_True_float16 (__main__.TestNN) ... skip: MIOpen SolverNotFound for FP32/Fp16/BF16 NHWC batchnorm SWDEV-509640 (0.001s)
  test_batchnorm_nhwc_eval_mixed_True_float32 (__main__.TestNN) ... skip: mixed precision is useless for float32 (0.001s)
----------------------------------------------------------------------
Ran 6 tests in 0.008s

OK (skipped=6)
```

shoud I switch to MIOpen Batchnorm API v2 in this PR?
should this test be adapted for CUDA as well and replace them?
- [TestNN.test_batchnorm_cudnn_nhwc](https://github.com/ROCm/pytorch/blob/d3c942e263149b3faedb58e345472affdbc55b2b/test/test_nn.py#L4936) - train mode fp32 NHWC 
- [TestNN.test_batchnorm_nhwc_cuda](https://github.com/ROCm/pytorch/blob/d3c942e263149b3faedb58e345472affdbc55b2b/test/test_nn.py#L5129) - evaluation mode fp32 NHWC  
- [TestNNDeviceType/test_batchnorm_eval_mixed](https://github.com/ROCm/pytorch/blob/d3c942e263149b3faedb58e345472affdbc55b2b/test/test_nn.py#L11217) - evaluation mode fp16/bf16 mixed NCHW

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]
LGTM for internal_testing branch.
After ROCm 6.4 is released and in upstream CI, please raise this patch to upstream and get the new UTs to work on CUDA also.
